pkgs/development/compilers/ghc/8.10.2-binary.nix

```diff
@@ -361,7 +365,11 @@ stdenv.mkDerivation rec {
     homepage = "http://haskell.org/ghc";
     description = "The Glasgow Haskell Compiler";
     license = lib.licenses.bsd3;
-    platforms = ["x86_64-linux" "armv7l-linux" "aarch64-linux" "i686-linux" "x86_64-darwin"];
+    # HACK: since we can't encode the libc / abi in platforms, we need
+    # to make the platform list dependent on the evaluation platform
+    # in order to avoid eval errors with musl which supports less
+    # platforms than the default libcs (i. e. glibc / libSystem)
```

Comment:
Sorry to bother you about this again, but maybe it would make things clearer if we mention here that we are doing this for the benefit of hydra, so `pkgsMusl.haskell.compiler.ghc8102Binary` can be evaluated properly even with darwin in `supportedSystems`.

(Technically only if hydra is not running on darwin)

Reply:
Sure, I'm happy to make the comment as good as possible -- but you understand the Hydra parts much better than me. If you could make a Github suggestion of how to word it, I'll apply it.

Reply:
```suggestion
    # platforms than the default libcs (i. e. glibc / libSystem).
    # This is done for the benefit of Hydra, so `packagePlatforms`
    # won't return any platforms that would cause an evaluation
    # failure for `pkgsMusl.haskell.compiler.ghc8102Binary`, as
    # long as the evaluator runs on a platform that supports
    # `pkgsMusl`.
```

Reply:
Done.